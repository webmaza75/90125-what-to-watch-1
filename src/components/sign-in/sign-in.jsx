import React, {Fragment} from 'react';
import PropTypes from 'prop-types';

import Header from '../header/header.jsx';
import PreHeader from '../pre-header/pre-header.jsx';
import Footer from '../footer/footer.jsx';

const SignIn = (props) => {
  const {
    email,
    password,
    onSubmit,
    onChangeEmail,
    onChangePassword,
    error,
    errorPassword
  } = props;

  const errorMassage = errorPassword ? errorPassword : error;

  return <Fragment>
    <PreHeader />
    <div className="user-page">
      <Header
        isRequiredAuthentication={true}
        pageType={`signIn`}
      />

      <div className="sign-in user-page__content">
        <form className="sign-in__form" onSubmit={(event) => onSubmit(event)}>
          {errorMassage && <div className="sign-in__message">
            <p>{errorMassage}</p>
          </div>}
          <div className="sign-in__fields">
            <div className="sign-in__field">
              <input
                className="sign-in__input"
                type="email"
                placeholder="Email address"
                name="user-email"
                id="user-email"
                value={email}
                onChange={(event) => onChangeEmail(event)}
              />
              <label className="sign-in__label visually-hidden" htmlFor="user-email">Email address</label>
            </div>
            <div className="sign-in__field">
              <input
                className="sign-in__input"
                type="password"
                placeholder="Password"
                name="user-password"
                id="user-password"
                value={password}
                onChange={(event) => onChangePassword(event)}
              />
              <label className="sign-in__label visually-hidden" htmlFor="user-password">Password</label>
            </div>
          </div>
          <div className="sign-in__submit">
            <button className="sign-in__btn" type="submit">Sign in</button>
          </div>
        </form>
      </div>
      <Footer />
    </div>
  </Fragment>;
};

SignIn.propTypes = {
  email: PropTypes.string,
  password: PropTypes.string,
  onSubmit: PropTypes.func,
  onChangeEmail: PropTypes.func,
  onChangePassword: PropTypes.func,
  error: PropTypes.string,
  errorPassword: PropTypes.string
};

export default SignIn;
