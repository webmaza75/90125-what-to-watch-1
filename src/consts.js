export const ALL_GENRES = `All genres`;
export const BASE_URL = `https://es31-server.appspot.com`;
export const ValidationErrors = {
  INVALID_EMAIL: `child \"email\" fails because [\"email\" is not allowed to be empty]`,
  INVALID_PASSWORD: `child \"password\" fails because [\"password\" is not allowed to be empty]`,
  INVALID_EMAIL_AND_PASSWORD: `\"email\" and \"password\" are not allowed to be empty`,
};
export const RatingLevels = {
  AWESOME: `Awesome`,
  VERY_GOOD: `Very good`,
  GOOD: `Good`,
  NORMAL: `Normal`,
  BAD: `Bad`
};
export const MAX_GENRES = 10;
export const MAX_FILMS = 20;
export const tabs = [
  `Overview`,
  `Details`,
  `Reviews`
];
export const Tabs = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};
export const NO_COMMENTS = `No comments`;
