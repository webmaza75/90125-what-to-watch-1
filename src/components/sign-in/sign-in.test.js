import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';

import {SignIn} from './sign-in.jsx';
import {ValidationErrors} from '../../consts.js';

describe(`SignIn correctly renders`, () => {
  it(`SignIn correctly renders without any errors`, () => {
    const signIn = renderer
      .create(<MemoryRouter initialEntries={[`/`]}>
        <SignIn
          email={`qwe@qwe.ru`}
          password={`123`}
          onFormSubmit={jest.fn()}
          onChangeEmail={jest.fn()}
          onChangePassword={jest.fn()}
          validationError={null}
          onSetError={jest.fn()}
        />
      </MemoryRouter>)
      .toJSON();

    expect(signIn).toMatchSnapshot();
  });

  it(`SignIn correctly renders with an error`, () => {
    const signIn = renderer
      .create(<MemoryRouter initialEntries={[`/`]}>
        <SignIn
          email={`qwe@qwe.ru`}
          password={` `}
          onFormSubmit={jest.fn()}
          onChangeEmail={jest.fn()}
          onChangePassword={jest.fn()}
          validationError={ValidationErrors.INVALID_PASSWORD}
          onSetError={jest.fn()}
        />
      </MemoryRouter>)
      .toJSON();

    expect(signIn).toMatchSnapshot();
  });
});
