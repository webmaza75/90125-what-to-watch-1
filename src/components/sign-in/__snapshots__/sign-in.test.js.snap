// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn correctly renders SignIn correctly renders with an error 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link "
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      Sign in
    </h1>
  </header>
  <div
    className="sign-in user-page__content"
  >
    <form
      className="sign-in__form"
      onSubmit={[Function]}
    >
      <div
        className="sign-in__message"
      >
        <p>
          child "password" fails because ["password" is not allowed to be empty]
        </p>
      </div>
      <div
        className="sign-in__fields"
      >
        <div
          className="sign-in__field"
        >
          <input
            className="sign-in__input"
            id="user-email"
            name="user-email"
            onChange={[MockFunction]}
            placeholder="Email address"
            type="email"
            value="qwe@qwe.ru"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-email"
          >
            Email address
          </label>
        </div>
        <div
          className="sign-in__field"
        >
          <input
            className="sign-in__input"
            id="user-password"
            name="user-password"
            onChange={[MockFunction]}
            placeholder="Password"
            type="password"
            value=" "
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-password"
          >
            Password
          </label>
        </div>
      </div>
      <div
        className="sign-in__submit"
      >
        <button
          className="sign-in__btn"
          type="submit"
        >
          Sign in
        </button>
      </div>
    </form>
  </div>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;

exports[`SignIn correctly renders SignIn correctly renders without any errors 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link "
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      Sign in
    </h1>
  </header>
  <div
    className="sign-in user-page__content"
  >
    <form
      className="sign-in__form"
      onSubmit={[Function]}
    >
      <div
        className="sign-in__fields"
      >
        <div
          className="sign-in__field"
        >
          <input
            className="sign-in__input"
            id="user-email"
            name="user-email"
            onChange={[MockFunction]}
            placeholder="Email address"
            type="email"
            value="qwe@qwe.ru"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-email"
          >
            Email address
          </label>
        </div>
        <div
          className="sign-in__field"
        >
          <input
            className="sign-in__input"
            id="user-password"
            name="user-password"
            onChange={[MockFunction]}
            placeholder="Password"
            type="password"
            value="123"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-password"
          >
            Password
          </label>
        </div>
      </div>
      <div
        className="sign-in__submit"
      >
        <button
          className="sign-in__btn"
          type="submit"
        >
          Sign in
        </button>
      </div>
    </form>
  </div>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
